### ISSUE TITLE
Privacy, Security, Toastr, and more

### ISSUE THREAD

**Issue by GaryCornell** [rappasoft/laravel-5-boilerplate#740]
Running latest Laravel 5.4 and latest build of boilerplate First off thanks for the project, got me started on my own project and now am expanding. But have found several things that need attention.

May not be an issue now that you went from name to first and last but before that anyone could sign up with the name admin or administrator as example ( or duplicate anyone else already registered) because the validation rules didn't have unique as a rule. Now of course some people have the same name so this  is a difficult point so I created a display_name field in database and updated the code. I think just for the sake of privacy you must have a display_name or username cause most people will not want their real name visible especially if you ever intend to add some social functionality between users, IE support forums, tickets or messenger which I have implemented already. And of course rules must make this name unique.

I also suggest using a hook before validation to exclude a list (created by admin ) of user/display names that cannot be used to register such as admin, administrator, etc.. You can even make this a comma separated list in config. Whatever your fancy it is imperative for privacy and security.

I will be adding some pull requests when I am happy with my current add ins for this project such as the messenger I have created, settings for admin, themes, toastr and many other functions I have or will add as time provides.

I also implemented Toastr notifications pretty much using the current setup, by including the Toastr css and js you can change the current withFlash*

Example:
**Current usage**
return redirect('messenger')->withFlashSuccess(trans('messages.frontend.created'));
**Toastr version**
return redirect('messenger')->withFlashSuccess(Toastr::success(trans('messages.frontend.created'), 'Success', ['options']));

options are listed in toastr config

```php
<?php

return ['options' => ["progressBar" => true,
    "closeButton" => false,
    "debug" => false,
    "newestOnTop" => true,
    "positionClass" => "toast-top-right",
    "preventDuplicates"=> true,
    "onclick" => null,
    "showDuration" => 300,
    "hideDuration" => 1000,
    "timeOut" => 5000,
    "extendedTimeOut" => 1000,
    "showEasing" => "swing",
    "hideEasing"=> "linear",
    "showMethod" => "fadeIn",
    "hideMethod" => "fadeOut"]
    ];
```
it can be implemented with minimal edits, works great.

**Messenger with Summernote**
![Image of Messenger](http://i249.photobucket.com/albums/gg226/PlanetMaster/messenger_zpsyiyvm3kc.png)

![Image of Comppose](http://i249.photobucket.com/albums/gg226/PlanetMaster/compose_zpshddwcu5i.png)

Includes sent, important, drafts, trash and custom save folders. Supports multiple user conversations and attachments. Excellent for user support functions.

Also adding in lots of nodejs and ajax for real time (chat, notifications, etc) and interactive functionality. Honestly have a ton more to post but this is a good start, hopefully will be active in helping improve the project.

Great project I hope to add to it, thanks

**Comment by GaryCornell** [comment]
Just to expand on the Toastr, I kept the includes.partials.messages blade and updated accordingly:

**current**
```php
@if ($errors->any())
    <div class="alert alert-danger">
        @foreach ($errors->all() as $error)
            {!! $error !!}<br/>
        @endforeach
    </div>
```

Updated Toastr version
```php
@if ($errors->any())
    <div class="alert alert-danger">
        @foreach ($errors->all() as $error)
            {!! Toastr::error($error, 'Error', ['options']); !!} 
        @endforeach
    </div>
```
Line breaks can be removed as Toastr provides its own padding styling.

Just to demonstrate the easy usage. At the same time it can be refactored to remove the blades all together and use just the session data via controller. Much cleaner/efficient method which I am currently in the process of implementing.

**Comment by monocidar** [comment]
@GaryCornell , waiting forward to see your improvisation on project 👍 Regards

**Comment by GaryCornell** [comment]
@monocidar  been busy as hell but your comment reminded me of this post. I started on the private messenger app setup in my Git and will add in the Toastr tomorrow, of course I already upgraded to Smart Notifications that have sound and are nicer then the older Toastr so I will add in both. Wont do a pull till I am happy with the code and compatibility. My main goal is efficiency and ease of insertion in the boilerplate. Ill post code and examples and maybe a few that use this can help out on optimizing and improving before I Pull. Either way I will put up all the code I have and you guys can fool around with it. really not that involved. Thanks
